ci/go.mod

```diff
@@ -1,12 +1,12 @@
 module github.com/dagger/dagger/ci
 
-go 1.21.7
+go 1.22
```

Comment:
This can't be bumped yet. See https://github.com/dagger/dagger/pull/7109#issue-2248420383.

Reply:
Yup, just pushed a change to `1.21`. `1.21.7` is failing for me:
<img width="859" alt="image" src="https://github.com/dagger/dagger/assets/3342/765d97b7-e8ab-44f6-8172-2557f29ccab4">

How did this work for you?

Reply:
:thinking: that's weird. The format did change at some point, is your version of the go language server up to date?

That said, I think it's fine to have it set to 1.21.

Reply:
I found the issue. It was incorrect ordering in `$PATH`, which was picking up the wrong `gopls`. All fixed now!